File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        ans = 0
        max_cost = max(costs)
        freq = [0]*(max_cost+1)
        for cost in costs:
            freq[cost]+=1

        for price in range(1,max_cost+1):
            if freq[price]==0:
                continue
            cnt = min(freq[price],coins//price)  
            ans+=cnt
            coins-=cnt*price
            if coins<price:
                break

        return ans               

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        int cnt = 0;
        int n = costs.size();
        sort(costs.begin(),costs.end());
        int i = 0;
        while(i<n && coins>0){
            coins-=costs[i];
            if(coins>=0)cnt++;
            i++;
        }
        return cnt;
    }
};